### ISSUE TITLE
ACES Output Transform Blink won't compile in OpenCL

### ISSUE THREAD

**Issue by nick-shaw** [jedypod/nuke-colortools#6]
The Blink versions of the ACES Output Transform (and quite possibly others of your Blink based nodes) won't run on the GPU on my MacBook Pro.

I can fix it by changing the `process()` function in the kernel to use this additional step:
```
SampleType(src) input = src();
float3 aces = float3(input.x, input.y, input.z);
```
Your original code referencing `src()` directly gives an error.
```
float3 aces = float3(src().x, src().y, src().z);
```
This change obviously also needs to be reflected in the final line:
```
dst() = float4(aces.x, aces.y, aces.z, input.w);
```

**Comment by jedypod** [comment]
Hey Nick! 
Thanks for the bug report, and apologies for the massive delay. I've pushed a fix for this issue in the ACES v1.2 blink script nodes.

**Comment by nick-shaw** [comment]
No worries!

I've also recently had issues getting Blink to compile on my M1 Mac Studio. I had to change this code in your Blink for the red modifier to make it work:
```
    float knots[5] = { -w/2.0f,
                       -w/4.0f,
                       0.0f,
                       w/4.0f,
                       w/2.0f };
```
Previously it was a double, not a float. And OpenCL on the M1 Ultra GPU objected to mixing doubles and floats. I've made a habit of obsessively putting ".0f" on the end of everything!
